### PR TITLE
Custom origin for staking pallet calls

### PR DESCRIPTION
**Pull request by Ank4n**
Fixes https://github.com/paritytech/substrate/issues/12930
polkadot companion: https://github.com/paritytech/polkadot/pull/6444

I have left `set_invulnerables` to still require root origin since it seems not something that should ever need to be set unless during the initial phase of a chain. Please correct me if this assumption is incorrect.

### CODE REVIEW

File: frame/staking/src/pallet/mod.rs [hunk 1]
```diff
@@ -183,8 +183,11 @@ pub mod pallet {
 		#[pallet::constant]
 		type SlashDeferDuration: Get<EraIndex>;
 
-		/// The origin which can cancel a deferred slash. Root can always do this.
-		type SlashCancelOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// The origin which can manage critical staking operations.
+		///
+		/// Set of operations that needs StakingAdminOrigin: `set_config`, changing validator count,
+		/// force era changes, `force_unstake`, `cancel_deferred_slash`.
+		type StakingAdminOrigin: EnsureOrigin<Self::RuntimeOrigin>;
```
Comment (kianenigma) [inline frame/staking/src/pallet/mod.rs:190]: This is in the staking pallet, can just be called `AdminOrigin` 😆

**Review by kianenigma — APPROVED** [review]

**Review by gpestana — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

**Review by melekes — APPROVED** [review]

**Comment by Ank4n** [comment]
Kian raised an issue and I agree that lowering privilege for the critical staking operations like reducing validator count or setting minimum/maximum bounds needs more careful thought. I will make the changes to the PR to only allow slash cancelling and setting minimum commission via `StakingAdmin` origin.

**Review by kianenigma — CHANGES_REQUESTED** [review]
Needs a rework as per the comment above.

**Comment by Ank4n** [comment]
Updated PR: https://github.com/paritytech/substrate/pull/13018

**Pull request closed without merging**